### PR TITLE
AGATE-13 - determine preferred language and save to the user profile for email notifications

### PR DESCRIPTION
**Pull request by meek0**

### CODE REVIEW

File: agate-core/src/main/java/org/obiba/agate/domain/User.java [hunk 1]
```diff
@@ -54,6 +56,8 @@
 
   private DateTime lastLogin;
 
+  private String notificationLocale;
```
Comment (ymarcon) [inline agate-core/src/main/java/org/obiba/agate/domain/User.java]: Usage is not limited to notifications so you should call this property: preferredLanguage

File: agate-core/src/main/resources/i18n/messages_de.properties [hunk 2]
```diff
@@ -2,3 +2,92 @@ title = Your request cannot be processed
 subtitle = Sorry, an error has occurred.
 status = Status:
 message = Message:
+email.resetPassword.title={0} - Password Reset
```
Comment (ymarcon) [inline agate-core/src/main/resources/i18n/messages_de.properties]: Remove this German translation file (unless you speak German)

File: agate-web-model/src/main/protobuf/Agate.proto [hunk 3]
```diff
@@ -90,6 +91,7 @@ message UserDto {
   optional TimestampsDto timestamps = 12;
   repeated GroupApplicationDto groupApplications = 13;
   optional string lastLogin = 14;
+  optional string notificationLocale = 15;
```
Comment (ymarcon) [inline agate-web-model/src/main/protobuf/Agate.proto]: notificationLocale => preferredLanguage

File: agate-webapp/src/main/webapp/app/user/views/user-identification-info-form.html [hunk 4]
```diff
@@ -19,6 +19,13 @@
 <div form-input name="user.firstName" model="user.firstName" label="user.firstName"></div>
 <div form-input name="user.lastName" model="user.lastName" label="user.lastName"></div>
 
+<div class="form-group">
+  <label class="control-label">
+    <span translate>user.notificationLocale</span>
+  </label>
+  <select class="form-control" value="{{getCurrentLanguage}}" name="user.notificationLocale" ng-model="user.notificationLocale" ng-options="locale for locale in languages"></select>
```
Comment (ymarcon) [inline agate-webapp/src/main/webapp/app/user/views/user-identification-info-form.html]: notificationLocale => preferredLanguage

File: agate-webapp/src/main/webapp/app/user/views/user-identification-info-form.html [hunk 5]
```diff
@@ -19,6 +19,13 @@
 <div form-input name="user.firstName" model="user.firstName" label="user.firstName"></div>
 <div form-input name="user.lastName" model="user.lastName" label="user.lastName"></div>
 
+<div class="form-group">
+  <label class="control-label">
+    <span translate>user.notificationLocale</span>
```
Comment (ymarcon) [inline agate-webapp/src/main/webapp/app/user/views/user-identification-info-form.html]: notificationLocale => preferredLanguage

File: agate-webapp/src/main/webapp/app/user/views/user-view.html [hunk 6]
```diff
@@ -57,6 +57,10 @@ <h3 translate>user.general-info</h3>
           <td>{{user.role}}</td>
         </tr>
         <tr>
+          <th translate>user.notificationLocale</th>
+          <td>{{user.notificationLocale}}</td>
```
Comment (ymarcon) [inline agate-webapp/src/main/webapp/app/user/views/user-view.html]: notificationLocale => preferredLanguage

File: agate-webapp/src/main/webapp/i18n/en.json [hunk 7]
```diff
@@ -352,6 +354,7 @@
     "view": "View",
     "password": "Password",
     "resetPassword": "Reset password",
+    "notificationLocale": "Notification Locale",
```
Comment (ymarcon) [inline agate-webapp/src/main/webapp/i18n/en.json]: notificationLocale => preferredLanguage

**Comment by meek0** [comment]
notificationLocale was changed for preferredLanguage

**Comment by ymarcon** [comment]
Preferred language is not editable by a simple user: field is missing in the profile view and modal.

**Comment by ymarcon** [comment]
In the user/profile view/form, the preferred language name should be translated: "en" -> "English" (when interface is in English.

**Comment by ymarcon** [comment]
User preferred language is not used when login agate: apply "fr" to a user, login with this user -> the interface is in English.

**Comment by ymarcon** [comment]
Add locale property to openid token:
https://github.com/obiba/agate/blob/master/agate-core/src/main/java/org/obiba/agate/service/TokenUtils.java#L217

**Comment by meek0** [comment]
setting the locale to the user preferred language when login is confirmed

**Comment by meek0** [comment]
added locale field for userProfile's JSONObject (get and update)

**Pull request merged**